packages/eslint-plugin/src/rules/no-empty-styled-expression/index.ts

```diff
@@ -0,0 +1,53 @@
+import type { Rule } from 'eslint';
+import type { CallExpression } from 'estree';
+
+import { isStyledImportSpecifier } from '../../utils/styled-import';
+
+type RuleModule = Rule.RuleModule;
+
+const isEmptyStyledExpression = (node: CallExpression): boolean => {
+  const [firstArg] = node.arguments;
+  if (firstArg?.type === 'ObjectExpression') {
+    return firstArg.properties.length === 0;
+  }
+  return true;
```

Comment:
There could be something else passed that is not an `ObjectExpression` and we'd throw an error. I think it's better that we only return true here if the args are empty (`node.arguments.length === 0`) or if there's a single arg with something we know is empty.

Reply:
That's fair enough. I'm happy to add these changes. Could I also ask for context on what kind of objects can we expect to be passed into styled.div(), and the impact of what happens when those objects are passed in?